File: back_end/models/user_model.js
const mongoose = require('mongoose')
const Schema = mongoose.Schema

const UserSchema = new Schema({
    "email": {
        type: String,
        required: true,
        trim: true,
        unique: true
    },
    "password": {
        type: String,
        required: true,
        trim: true,
    },
    "name": {
        type: String,
        required: true,
        trim: true,
    },
    "surname": {
        type: String,
        required: true,
        trim: true,
    },
    "phoneNumber": {
        type: String,
        required: true,
        trim: true,
        unique: true
    },
    "countryCode": {
        required: true,
        trim: true,
        type: String,
    }
}, { collection: "user", versionKey: false })

const User = mongoose.model('User', UserSchema)
module.exports = User

File: mobil/reyclist_mobil/lib/ui/login/model/user_model.dart
import 'package:json_annotation/json_annotation.dart';

part 'user_model.g.dart';

@JsonSerializable()
class User {
  const User({
    this.id,
    this.email,
    this.password,
    this.name,
    this.surname,
    this.phoneNumber,
    this.countryCode,
    this.point,
  });

  final String? id;
  final String? email;
  final String? password;
  final String? name;
  final String? surname;
  final String? phoneNumber;
  final String? countryCode;
  final int? point;

  factory User.fromJson(Map<String, dynamic> json) {
    return _$UserFromJson(json);
  }

  Map<String, dynamic>? toJson() {
    return _$UserToJson(this);
  }
}
